auth/realm/base/src/test/java/org/wildfly/security/auth/realm/AggregateRealmEvidenceTest.java

```diff
@@ -0,0 +1,328 @@
+        SecurityRealm authenticationRealm = toSecurityRealm(authentication, "authentication", IDENTITY_PRINCIPAL);
+        SecurityRealm[] authorizationRealms = new SecurityRealm[authorization.length];
+        for (int i = 0; i < authorizationRealms.length; i++) {
+            if (authentication == authorization[i]) {
+                authorizationRealms[i] = authenticationRealm;
+            } else {
+                authorizationRealms[i] = applyPrincipalTransformer ? toSecurityRealm(authorization[i], "authorization"+String.valueOf(i), IDENTITY_PRINCIPAL_TRANSFORMED) : toSecurityRealm(authorization[i], "authorization"+String.valueOf(i), IDENTITY_PRINCIPAL);
+            }
+        }
+
+        return new AggregateSecurityRealm(authenticationRealm, principalTransformer, authorizationRealms);
+    }
+
+    private SecurityRealm toSecurityRealm(Attributes attributes, String path, Principal principal) throws Exception {
+        SecurityRealm securityRealm = new FileSystemSecurityRealm(getRootPath(path, true));
+        ModifiableRealmIdentity realmIdentity;
+
+        if(attributes != null) {
+            realmIdentity = ((FileSystemSecurityRealm) securityRealm).getRealmIdentityForUpdate(principal);
+            realmIdentity.create();
+            realmIdentity.setAttributes(attributes);
+            realmIdentity.dispose();
+            return securityRealm;
+        }
+        return securityRealm;
+    }
+
+    /*
+     * Function to convert string to all caps
+     */
+    private class CaseRewriter implements NameRewriter {
+        public String rewriteName(String original) {
+            return (original == null) ? null : original.toUpperCase(Locale.ROOT);
+        }
+    }
+
+    private static X509Certificate[] populateCertificateChain(boolean includeSubjectAltNames) throws Exception {
+        KeyPairGenerator keyPairGenerator;
+        try {
+            keyPairGenerator = KeyPairGenerator.getInstance("RSA");
+        } catch (NoSuchAlgorithmException e) {
+            throw new Error(e);
+        }
+        final KeyPair[] keyPairs = new KeyPair[5];
+        for (int i = 0; i < keyPairs.length; i++) {
+            keyPairs[i] = keyPairGenerator.generateKeyPair();
+        }
+        final X509Certificate[] orderedCertificates = new X509Certificate[1];
+        for (int i = 0; i < orderedCertificates.length; i++) {
+            X509CertificateBuilder builder = new X509CertificateBuilder();
+            X500PrincipalBuilder principalBuilder = new X500PrincipalBuilder();
+            principalBuilder.addItem(X500AttributeTypeAndValue.create(X500.OID_AT_COMMON_NAME,
+                    ASN1Encodable.ofUtf8String("bob" + i)));
+            X500Principal dn = principalBuilder.build();
+            builder.setSubjectDn(dn);
+            if (i == orderedCertificates.length - 1) {
+                // self-signed
+                builder.setIssuerDn(dn);
+                builder.setSigningKey(keyPairs[i].getPrivate());
+            } else {
```

Comment:
@ashley-abdelsayed98: length of orderedCertificates array is always 1, so this else branch is never used, [this](https://github.com/wildfly-security/wildfly-elytron/pull/1296/commits/0115b07451aa3616d63e3566610523705939bab7#diff-cf27ec2069feba14e254d5bf7dacd186R273) for loop is always used only once and generation of 5 keyPairs is useless as well (only one is used). Method attribute `includeSubjectAltNames` is never used as well. Is there some typo (and the fix of this typo fixes also this issue)?

Reply:
Sorry this implementation was copied from another test. Fixed now to be the correct implementation for 1 keyPair